### ISSUE TITLE
Allow user to choose left or right side for ambiguous times

### ISSUE THREAD

**Issue by ConnorLanglois** [moment/luxon#1397]
**Is your feature request related to a problem? Please describe.**
First off, nice library, seems more put together than others I have looked at. But I am concerned that it's knowingly accepted that the offset used for dates that are ambiguous (i.e. when dst causes duplicate hour) is undefined. Doesn't this make any local date calculation whatsoever with this library totally unreliable?

**Describe the solution you'd like**
Adding a chained method to the DateTime, such as `ifAmbiguousChoose` with positional argument `side: 'left' | 'right'`. Or something like that. You can doing the probing here to determine which side the DateTime lies on and then choose one based on the argument supplied.

**Describe alternatives you've considered**
Looking at issue #914, I see there has been some discussion around this and the argument of performance concerns. But if this optional method were added, then users would knowingly accept the performance from using it. Generally, correctness is preferred to performance when it comes to fundamental mission-critical things like datetimes.

**Comment by diesieben07** [comment]
Unfortunately this is not really possible to implement without Luxon shipping its own time zone data. There is no browser API to check whether or not a given local date & time is ambiguous. As such Luxon does not know whether a given time is ambiguous, it can only check whether its valid.

If all DST changes were 1 hour, we could "just" check one hour previous to see if the local wall time stays the same - but that only works for most cases and there is no way for Luxon to detect whether the check was correct.
If you have an idea for how to implement this method using the APIs available in browsers, I would be happy to work on a PR.

**Comment by ConnorLanglois** [comment]
I think it should be possible, as long as a somewhat loose assumption is accepted (the ±1 day), see below. Look at [how the temporal polyfill is accomplishing this here](https://github.com/js-temporal/temporal-polyfill/blob/c0f7349a327b68543797d38e045b1fd8c1e0949b/lib/ecmascript.ts#L3063). This function takes the timezone *(id)* and the date parts *(year, month day, ...)* and returns an array of possible utc epoch nanosecond values. Looking at it for awhile, it seems that what they are doing is this, using example datetime '2023-11-05 01:30:00 America/New_York' *(ambiguous time when switching from EDT back to EST)*:

**1.)** Take datetime, throw away the timezone for now, and convert that to epoch nanosecond; this represents some utc instant. We'll call this **ns**.
**2a.)** Subtract a day's worth of nanoseconds from **ns**. Give this + the timezone *(America/New_York)* to the browser and calculate offset here. We get -04:00.
**2b.)** Add a day's worth of nanoseconds from **ns**. Give this + the timezone *(America/New_York)* to the browser and calculate the offset here. We get -05:00.
**3.)** For each offset:
**3a.)** Subtract the offset *(in nanoseconds)* from **ns**. This gives us the utc equivalent of the original datetime *(2023-11-05 01:30:00)* at this particular offset *(e.g. -04:00)*. Remember in our example the offset is negative, so subtracting here will add.
**3b.)** Give this nanosecond value + our timezone *(America/New_York)* to the browser and ask it to give us the date parts *(year, month, day, ...)*.
**3c.)** If all the date parts *(year, month, day, hour, minute, ...)* equal the original datetime *(2023-11-05 01:30:00)*, then we have a match and this is one possible value.

<details>
<summary>Why is step 3c needed?</summary>

I believe this check in step 3c is needed because in step 2 we subtracted and added 1 day, which isn't some precise value. It seems somewhat arbitrary to me *(but perhaps was chosen to make sure that ±1 day = 2 days encompasses [the max timezone difference](https://stackoverflow.com/a/8131056) and is not too large (> ±1 day) in case multiple dst switches for some reason occurred in a short timespan)*, and because of this it's possible that the date parts won't match the original datetime here.

For instance, if our original example was at time 00:00, then even though subtracting a day would give offset -04:00 and adding a day would give -05:00, -05:00 wouldn't be a possible match because the *hour* date part in step 3b that we get from the browser would be 01:00, not the original 00:00. Here is some Luxon code to verify this math:
```js
> DateTime.fromISO('2023-11-05', { zone: 'UTC' }).minus({ hours: -5 }).setZone('America/New_York').toString()
'2023-11-05T01:00:00.000-04:00'
```
</details>

<details>
<summary>Here is some sample code to check if DateTime is ambiguous using this algorithm</summary>

This code uses Luxon since it makes things easier and more intuitive without nanosecond math:
```js
const { DateTime } = require("luxon");

const dateToCheck = '2023-11-05T01:30:00';
const timezone = 'America/New_York';

const date = DateTime.fromISO(dateToCheck, { zone: timezone });
const dateWithoutTimezone = date.setZone('UTC', { keepLocalTime: true }); // step 1

const earlierOffset = dateWithoutTimezone.minus({ days: 1 }).setZone(timezone).offset; // step 2a
const laterOffset = dateWithoutTimezone.plus({ days: 1 }).setZone(timezone).offset; // step 2b
const found = earlierOffset === laterOffset ? [earlierOffset] : [earlierOffset, laterOffset];

const dateParts = date.toObject();
const possibleDates = found.map(offset => {
  const possibleDate = dateWithoutTimezone.minus({ minutes: offset }).setZone(timezone); // steps 3a and 3b
  const possibleDateParts = possibleDate.toObject();
  const areDatesPartsEqual = Object.keys(dateParts).every(key => dateParts[key] === possibleDateParts[key]); // step 3c
  if (areDatesPartsEqual) {
    return possibleDate;
  } else {
    return undefined;
  }
  return possibleDate;
}).filter(d => d);

console.log(possibleDates.map(d => d.toISO()));
```
</details>

**Comment by diesieben07** [comment]
I agree. My idea would be the leave the current behavior as is (i.e. keep it ambiguous) and let people opt-in to the non-ambiguous behavior if they want to.
The simplest option imo would be a method on `DateTime` like `ifAmbiguousChoose('later'|'earlier')` (hand-wavy name), but it might be possible to instead use a setting or an option on existing methods, provided that doesn't introduce too much complexity there.
